src/components/Table/Table.tsx

```diff
@@ -0,0 +1,186 @@
+        setCurrentFilters((prev) => ({
+            ...prev,
+            [key]: value,
+        }));
+    };
+
+    // Apply filters using predicate functions
+    let filteredData = data;
+    if (filters) {
+        filteredData = data.filter((item) => {
+            const filterKeys = Object.keys(filters) as FilterKey[];
+
+            return filterKeys.every((filterKey) => {
+                const filterConfig = filters[filterKey];
+                const filterValue = currentFilters[filterKey];
+
+                // If filter value is empty/undefined, include the item
+                if (filterValue === undefined || filterValue === null) {
+                    return true;
+                }
+
+                // Handle multi-select filters (array values)
+                if (filterConfig.filterType === 'multi-select') {
+                    const filterValueArray = Array.isArray(filterValue) ? filterValue.filter((v): v is string => typeof v === 'string') : [];
+                    if (filterValueArray.length === 0) {
+                        return true;
+                    }
+                    // For multi-select, pass the array of selected values
+                    return isItemInFilter?.(item, filterValueArray) ?? true;
+                }
+
+                // Handle single-select filters
+                const singleValue = typeof filterValue === 'string' ? filterValue : '';
+                return singleValue === '' || (isItemInFilter?.(item, [singleValue]) ?? true);
+            });
+        });
+    }
+
+    const [activeSearchString, updateSearchString] = useState('');
+
+    // Apply search using onSearch callback
+    let filteredAndSearchedData = filteredData;
+    if (isItemInSearch && activeSearchString.trim()) {
+        filteredAndSearchedData = filteredData.filter((item) => isItemInSearch(item, activeSearchString));
+    }
+
+    const sortToggleCountRef = useRef(0);
+    const [activeSorting, setActiveSorting] = useState<ActiveSorting<ColumnKey>>({columnKey: undefined, order: 'asc'});
+
+    const updateSorting: UpdateSortingCallback<ColumnKey> = ({columnKey, order}) => {
+        if (columnKey) {
+            setActiveSorting({columnKey, order: order ?? 'asc'});
+            return;
+        }
+
+        setActiveSorting({columnKey: undefined, order: 'asc'});
+    };
+
+    const toggleSorting: ToggleSortingCallback<ColumnKey> = (columnKey) => {
+        if (!columnKey) {
+            updateSorting({columnKey: undefined});
+            sortToggleCountRef.current = 0;
+            return;
+        }
+
+        setActiveSorting((currentSorting) => {
+            if (columnKey !== currentSorting.columnKey) {
+                sortToggleCountRef.current = 0;
+                return {columnKey, order: 'asc'};
+            }
+
+            // Check current toggle count to decide if we should reset
+            if (sortToggleCountRef.current >= MAX_SORT_TOGGLE_COUNT) {
```

Comment:
I don't really understand the toggle count or why it is needed. Why can't it just have the toggle sort order logic from a few lines below?